src/transport_picoquic.cpp

```diff
@@ -730,23 +741,27 @@ PicoQuicTransport::close([[maybe_unused]] const TransportContextId& context_id)
 void
 PicoQuicTransport::checkTxData()
 {
-    for (auto& c_pair : active_streams) {
-        for (auto& s_pair : active_streams[c_pair.first]) {
+    // std::map<TransportContextId, std::map<StreamId, StreamContext>> active_streams;
+    for (auto& [tcid, streams] : active_streams) {
+        for (auto& [sid, stream] : streams) {
 
-            if (s_pair.first != 0) {
-                if (s_pair.second.tx_data->size() > 0)
+            if (sid != 0) {
+                // QUIC Stream
+                if (stream.tx_data->size() > 0)
                     (void)picoquic_mark_active_stream(
-                      s_pair.second.cnx, s_pair.first, 1, static_cast<StreamContext*>(&s_pair.second));
+                      stream.cnx, sid, 1, static_cast<StreamContext*>(&stream));
             } else {
-                if (s_pair.second.tx_data->size() > 0) {
+                // QUIC Datagram
+                if (stream.tx_data->size() > 0) {
                     // reinsert by wake is one way to get picoquic to callback sooner
                     // picoquic_reinsert_by_wake_time(s_pair.second.cnx->quic, s_pair.second.cnx, cur_time);
                     // Only send tx data here when using picoquic queueing, otherwise let the callbacks handle it
                     // sendTxData(&s_pair.second, NULL, 1400);
-
-                    (void)picoquic_mark_datagram_ready(s_pair.second.cnx, 1);
+                    //picoquic_mark_datagram_ready(stream.cnx, 1);
+                    update_datagram_ready_status(stream.cnx, true);
                 } else {
-                    (void)picoquic_mark_datagram_ready(s_pair.second.cnx, 1);
+                    //picoquic_mark_datagram_ready(stream.cnx, 0);
+                    update_datagram_ready_status(stream.cnx, false);
```

Comment:
Seems like both the if and else statement are practically identical? Maybe instead:
```suggestion
update_datagram_ready_status(stream.cnx, stream.tx_data->size() > 0);
```
The comment inside the if block could remain to provide context.

Reply:
In the new changes if and else are doing opposite logic. Yes, the old code was doing exactly the same thing and causing more busy loop calls and new change should address it

Reply:
+1 for ```update_datagram_ready_status(stream.cnx, stream.tx_data->size() > 0);```